src/sentry/event_manager.py

```diff
@@ -559,6 +559,26 @@ def save_error_events(
         ), metrics.timer("event_manager.calculate_event_grouping", tags=metric_tags):
             hashes = _calculate_primary_hash(project, job, grouping_config)
 
+        if secondary_hashes:
+            tags = {
+                "primary_config": grouping_config["id"],
+                "secondary_config": secondary_grouping_config["id"],
+            }
```

Comment:
> This is potentially unbounded. Also every **combination** of tag values counts towards cardinality, so it's only going to get worse once there are multiple secondaries.

It's a valid concern, but I _claim_ this is not going to be too bad. Here's what I'm thinking:

- The point of this metric is to be able to answer the question "How often does releasing a new config change lead to new hashes?"

- Once all is said and done, project-specific configs will be a thing of the past, which means we'll only ever be comparing the current config to the immediately previous config, so there'll only be as many combos as there have been config transitions. Given that it's month by month (TIL!), that's likely to  be a reasonably-sized number. (Note that even if a given group only sees events under configs `A` and `C` (`C` being the current config), we'll still have to calculate the hash under config `B`, and will therefore still be able to compare it to config `C`'s hash.) 

- In the meantime, the metric is (sort of) meaningless, for two reasons:
  - Some of our comparisons will be between hashes calculated by non-consecutive configs.
  - Even when the two configs _are_ consecutive, our configs aren't yet immutable, so two given config `X` to config `Y` comparisons might not actually even mean quite the same thing.

- Given that, we could just hold off on sending the metric entirely, or we could only send it when the secondary config is whatever one comes right before the current one. I'm fine with either of these scenarios, though for interest's sake I feel like we might as well send it now, so long as the cost isn't crazy.

- If we do send it now, as is, the cardinality should be low because we just don't have that many configs. (Five, to be specific.) Calling them `A` - `E` for simplicity:
  - `E` as a primary config can be preceded by any of the other 4 configs
  - `D` as a primary config can be preceded by any of `A`, `B`, or `C`
  - `C` as a primary config can be preceded by either of `A` or  `B`
  - `B` as a primary config can only be preceded by `A`

  So there are 10 total combos right now (or actually, 9, because there are no active projects with `B` as their primary).

- Between now and the eventual state, there will be a time during which project-specific configs aren't yet gone, but don't represent either of the two most recent configs we'll be checking. Once that happens (in other words, once we release the second universal config), we'll already be in a world where we'll only be comparing to the immediately-previous config, which puts us into the as-many-combos-as-configs-in-a-month scenario.

So, I think the options are:

1) Merge this as-is, in which case there are 9 combos until there are the same number of convos as config releases that month.
2) Change this to only record the metric if the secondary is the one right before the current one, in which case there's 1 combo until there are the same number of convos as config releases that month.
3) Change this so the metric is temporarily off entirely, in which case there are 0 combos until there are the same number of convos as config releases that month.

My gut says 1) is still not too expensive, though I can tone it down to 2) or 3) if you think it's better.

Thoughts?

Reply:
my feeling is that this is borderline acceptable for now but needs revisiting for multiple secondaries

Reply:
Sounds reasonable to me. FWIW, when I initially commented I was under impression that it might be whole 15 months worth of combinations (i.e. full DD retention period), but according to the billing doc I've linked it's only current month, so indeed it ought to be fine.